src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -230,18 +230,9 @@ private static int[] AdjustDimensions(OnnxShape shape)
             // it will be runtime exception, util Maps and Sequences become supported.
             if (shape.Count > 0)
             {
-                // some models may have -1 in first position.
-                // skip this dimension when setting output column dimensions.
-                if (shape[0] < 0)
-                {
-                    return shape.Skip(1).Select(x => (int)x).ToArray();
-                }
-                else
-                {
-                    return shape.Select(x => (int)x).ToArray();
-                }
+                return shape.Select(x => (x <= 0) ? 1 : x).ToArray();
```

Comment:
If the dimension is 0, then you return 1 here? Should it just be ‘x < 0 ? 1 : x’

Reply:
Onnxruntime will throw an exception if any of the dimensions of a tensor are actually zero (as will SNT).  

The code above is attempting to be friendly to size calculations as well (which we do later on). For example, if some converted model has trailing zeros (hypothetically), such as  [3,5,5,0,0,0,0], the size of the buffer we need to provide is more likely to be 75  than 0 (and the zeros in the dimension values are  most likely due to a badly converted model). 

Converting the zero to a 1 seems like a safer option than blocking execution.